skiko/src/commonTest/kotlin/org/jetbrains/skia/FontFallbackTest.kt

```diff
@@ -0,0 +1,53 @@
+package org.jetbrains.skia
+
+import org.jetbrains.skia.impl.use
+import org.jetbrains.skia.paragraph.FontCollection
+import org.jetbrains.skia.paragraph.TypefaceFontProvider
+import org.jetbrains.skia.tests.makeFromResource
+import org.jetbrains.skiko.kotlinBackend
+import org.jetbrains.skiko.tests.runTest
+import kotlin.test.Test
+import kotlin.test.assertContentEquals
+import kotlin.test.assertEquals
+import kotlin.test.assertNotEquals
+
+class FontFallbackTest {
+
+    @Test
+    fun testChinese_withFallback() = runTest {
+        val fm = TypefaceFontProvider.createAsFallbackProvider()
+        val notoSansSC = Typeface.makeFromResource("./fonts/NotoSansSC-Regular.ttf", 0)
+        fm.registerTypeface(notoSansSC)
+
+        FontCollection().use {
+            it.setDefaultFontManager(FontMgr.defaultWithFallbackFontProvider(fm))
+
+            // 0x6C34 = "水"
+            val df = it.defaultFallback(0x6C34, FontStyle.NORMAL, null)
+            val glyphs = df!!.getStringGlyphs("水")
+
+            if (kotlinBackend.isWeb()) {
+                assertEquals("Noto Sans SC", df.familyName)
+                assertContentEquals(shortArrayOf(16391), glyphs)
+            } else {
+                // the actual font can be different on different systems,
+                // so just making sure it's not null
+                assertNotEquals(null, df?.familyName)
```

Comment:
looks like safe-call is redundant here

Reply:
I agree with you. But the compiler doesn't like it:

> Type inference failed. The value of the type parameter T should be mentioned in input types (argument types, receiver type or expected type). Try to specify it explicitly.


If I add the type explicitly like this: `assertNotEquals<String?>(null, df.familyName)` 
it says that I can remove explicit type arguments. When I do, it shows the error as before.

Looks like a bug.

Reply:
changed to explicit type and removed a safe-call